src/transformStateWithClones.js

```diff
@@ -1,13 +1,51 @@
 'use strict';
 
+function transform(prevState, action) {
+  switch (action.type) {
+    case 'addProperties':
+      return {
+        ...prevState,
+        ...action.extraData,
+      };
+
+    case 'removeProperties': {
+      const newState = { ...prevState };
+
+      for (const key of action.keysToRemove) {
+        if (Object.keys(newState).includes(key)) {
```

Comment:
if you try to access a non-existing property and then delete it, nothing will happen, thus this validation is simply redundant